pkg/beacon/analyzer.go

```diff
@@ -288,3 +303,121 @@ func countAndRemoveConsecutiveDuplicates(numberList []int64) ([]int64, map[int64
 	}
 	return result, counts
 }
+
+//getTsHistogramScore
+func getTsHistogramScore(min int64, max int64, tsList []int64) ([]int64, []int, map[int]int, float64) {
+
+	// get bucket list
+	// we currently look at a 24 hour period
```

Comment:
If we want to cover those cases, we need to make adjustments to not assume a 24hr period in other places (like the frontend) as well. The current intention is to look at a 24hr period so that's something we should look at carefully and make consistent across the board

Reply:
I agree this is a large issue we need to consider on the whole. 

Please add a note here that while the intent is to look at a 24 buckets with each bucket representing an hour, we may be looking at a smaller or larger time period.